src/ir/ExpressionManipulator.cpp

```diff
@@ -80,10 +80,11 @@ flexibleCopy(Expression* original, Module& wasm, CustomCopier custom) {
     }
     Expression* visitCallIndirect(CallIndirect* curr) {
       auto* ret = builder.makeCallIndirect(
-        curr->fullType, copy(curr->target), {}, curr->type, curr->isReturn);
+        copy(curr->target), {}, curr->sig, curr->isReturn);
       for (Index i = 0; i < curr->operands.size(); i++) {
         ret->operands.push_back(copy(curr->operands[i]));
       }
+      ret->type = curr->type;
```

Comment:
Is this needed because `makeCallIndirect` no longer receives the `type`? I think perhaps it should still receive it. We don't want to risk people forgetting to set it after calling (`make*` should return a fully-formed node), and in general, where the type can't be inferred from the inputs, we add the type as an extra input (e.g. `makeLocalGet`, `makeCall`).

Reply:
`makeCallIndirect` performs finalization, which fills out the `type` using the passed `Signature`. The problem here is that if any of the operands is unreachable, the `type` should also be unreachable, but since the operands are added after finalization we need to propagate that information manually.

The code would be cleaner if the operands were passed as part of the builder call so finalization can take care of this. I'll make that change.